source/linux/cpu_usage_sampler.c

```diff
@@ -53,39 +61,40 @@ static void aws_get_cpu_sample_fn_linux_get_uint64_delta(uint64_t first, uint64_
     }
 }
 
-static int aws_get_cpu_sample_fn_linux(struct aws_cpu_sampler *sampler, double *output) {
-    struct aws_cpu_sampler_linux *sampler_linux = sampler->impl;
+static int aws_get_cpu_sample_fn_linux(struct aws_system_cpu_sampler *sampler, double *output) {
+    struct aws_system_cpu_sampler_linux *sampler_linux = sampler->impl;
 
     int return_result = AWS_OP_ERR;
-    uint64_t total_user, total_user_low, total_system, total_idle, total;
-    s_get_cpu_usage_linux(&total_user, &total_user_low, &total_system, &total_idle);
-    // total_combined needs to be double to allow for fractions
-    double percent, total_combined;
-
+    // Have to define these here for cleanup
+    uint64_t total_user = 0, total_user_low = 0, total_system = 0, total_idle = 0;
```

Comment:
the way it is now, if this function fails, we end up storing `sampler_linux->cpu_last_total_user = 0` before returning.

It's generally a best-practice to leave the state of something unchanged if there's an error. The way it is now, if there's a 1-time failure the stored value will be zeroed out, and the next sample to succeed is going to look really weird because it will be a delta from 0.

You probably don't want to update ``sampler_linux->cpu_last_total_user` and friends unless the function has completely succeeded. You might be able to get away from using `goto` at all in this function, since it turns out there's nothing really to clean up, and just go with early-return on failure